phoenix-core/src/main/java/org/apache/phoenix/schema/MetaDataClient.java

```diff
@@ -3693,6 +3706,25 @@ public MutationState addColumn(PTable table, List<ColumnDef> origColumnDefs,
                                 .setTableName(tableName).build().buildException();
                             }
                             PColumn column = newColumn(position++, colDef, PrimaryKeyConstraint.EMPTY, table.getDefaultFamilyName() == null ? null : table.getDefaultFamilyName().getString(), true, columnQualifierBytes, willBeImmutableRows);
+                            HashMap<PTable, PColumn> indexColumn = null;
+                            if (cascade && !colDef.isPK() && numIndexes>0) {
```

Comment:
Perhaps just do and outer if on cascade to make it more readable and maintainable. 
if (cascade) {
     if (colDef.isPK) {
        throw Ex
    }
    for (PTable :) {
   }
}

Reply:
I forget why I had numIndexes -- doesnt make sense. Thank you for correction.